src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,22 +43,46 @@
 
     function load(templateId) {
       $scope.loadInProgress = true;
-
-      return templates.getTemplateWithBody(templateId).then(function (result) {
+      return templates.getTemplate(templateId).then(function (template) {
+        switch (template.bodyType) {
+          case "empty":
+            $scope.isHtmlTemplate = !$rootScope.forceMsEditor;
+            break;
+          case "mseditor":
+            $scope.isHtmlTemplate = false;
+            break;
+          case "rawHtml":
+            $scope.isHtmlTemplate = true;
+            break;
+        }
+        return {
+          from_name: template.from_name,
+          from_email: template.from_email,
+          subject: template.subject,
+          id: template.id,
+          name: template.name,
+          body: $scope.isHtmlTemplate? templates.getTemplateBody(templateId) : null
+        };
+      }).then(function (result) {
```

Comment:
Why did you choose using two _then_?

Reply:
To return only the value from getTemplateBody's promise and assign it in $scope.template.content

Reply:
ok, but why did you split it?